graphs/util/gpml_util_interface.py

```diff
@@ -0,0 +1,296 @@
+        # YET!!
+        interface.add_node_label(G, count, node['TextLabel'])
+
+        # Again not sure about font size
+        interface.add_node_fontsize(G, count, node['FontSize'])
+
+        if 'Color' in node:
+            interface.add_node_border_color(G, count, '#' + node['Color'])
+            interface.add_node_fill_color(G, count, '#' + node['Color'])
+        interface.add_node_color(G, count, 'white')
+        interface.add_node_background_opacity(G, count, 0)
+
+        # temp_node['data']['border_width'] = 2
+        # rectangle is the default shape for all the datanodes
+        interface.add_node_shape(G, count, 'rectangle')
+        # temp_node['data']['shape'] = 'rectangle'
+        # We use compound nodes to represent GPML groups in GraphSpace.
+        if 'GroupRef' in node:
+            interface.add_node_parent(G, count, node['GroupRef'])
+
+        # increase count, i.e. to keep unique ids for shapes, nodes, labels.
+        count = count + 1
+    return G, count
+
+
+def parse_labels(G,   count, graph_gpml):
+    # Parsing the label element for all the details.
+    labels =  graph_gpml.findall(GPML+'Label')
+
+    label_details = []
+    for label in labels:
+        temp_node = {}
+        temp_node.update(label.attrib)
+        for more_details in label:
+            temp_node.update(more_details.attrib)
+        label_details.append(temp_node)
+    for node in label_details:
+        temp_node = {'data': {}}
+        # Give an id to the label element.
+        interface.add_node(G, count)
+        _node_graphics(G, count, node)
+        # temp_node['data']['x'] = node['CenterX']
+        # temp_node['data']['y'] = node['CenterY']
+        interface.add_node_x_coordinate(G, count, node['CenterX'])
+        interface.add_node_y_coordinate(G, count, node['CenterY'])
+        # Again not sure about this
+        # temp_node['data']['content'] = node['TextLabel']
+        interface.add_node_label(G, count, node['TextLabel'])
+        # Again not sure about this
+        interface.add_node_fontsize(G, count, node['FontSize'])
+        # temp_node['data']['font_size'] = node['FontSize']
+        # By default we choose white :)
+        interface.add_node_color(G, count, 'white')
+        # temp_node['data']['background_color'] = 'white'
+        interface.add_node_background_opacity(G, count, 0)
+        # temp_node['data']['background_opacity'] = 0
+        # increase count, i.e. to keep unique ids for shapes, nodes, labels.
+        count = count + 1
+    return G,   count
+
+
+def parse_interactions(G,   count, interactions):
+    # Create a list of (x, y) coordinates of an edge
+    # every element of the list can be either of length of 2 or 3
+    # as curved edges are represented by 3 coordinates.
+    edge_details = []
+    for edge in interactions:
+        for details in edge:
+            temp_edge = []
+            for more in details:
+                if 'X' in more.attrib and 'Y' in more.attrib:
```

Comment:
You will need to parse the GraphRefs that correspond to the nodes connected by this interaction.